arviz/utils.py

```diff
@@ -50,14 +54,34 @@ def _var_names(var_names, data):
                 )
             )
 
+        excluded_vars = [
+            var[1:] for var in var_names if var.startswith("~") and var not in all_vars
+        ]
         if excluded_vars:
-            var_names = [i for i in all_vars if i not in excluded_vars]
-
-        existent_vars = np.isin(var_names, all_vars)
-        if not np.all(existent_vars):
+            if (filter == "like") or (filter == "regex"):
```

Comment:
filter in ("like, "regex")

We should also call `str(filter).lower()` before this line

Reply:
You sound just like pylint 😜